2d/oscillating_cylinder_new/tank.py

```diff
@@ -0,0 +1,592 @@
+
+if opts.circle2D==True:
+
+    vertices=[[0.0, 0.0],#0
+              [x1,  0.0],#1
+              [x2,  0.0],#2
+              [x3,  0.0],#3
+              [x4,  0.0],#4
+              [x5,  0.0],#5
+              [x5,  tank_dim[1]],#6
+              [x4,  tank_dim[1]],#7
+              [x1,  tank_dim[1]],#8
+              [0.0, tank_dim[1]],#9
+              ]
+
+    vertexFlags=np.array([1, 1, 1, 1, 1, 1,
+                          3, 3, 3, 3,
+                          ])
+    segments=[[0,1],
+              [1,2],
+
+              [2,3],
+
+              [3,4],
+              [4,5],
+              [5,6],
+              [6,7],
+              [7,8],
+              [8,9],
+              [9,0],
+
+              [1,8],
+              [4,7],
+             ]
+
+    segmentFlags=np.array([ 1, 1, 1, 1,   
+                            1,                        
+                            2, 3, 3, 3, 4,
+                            5, 5,
+                         ])
+
+
+regions = [ [ 0.90*x1 , 0.10*tank_dim[1] ],
+            [ 0.90*x2 , 0.10*tank_dim[1] ],
+            [ 0.95*tank_dim[0] , 0.95*tank_dim[1] ] ]
+
+regionFlags=np.array([1, 2, 3])
+
+
+
+tank = st.CustomShape(domain, vertices=vertices, vertexFlags=vertexFlags,
+                      segments=segments, segmentFlags=segmentFlags,
+                      regions=regions, regionFlags=regionFlags,
+                      boundaryTags=boundaryTags, boundaryOrientations=boundaryOrientations)
+
+
+
+############################################################################################################################################################################
+# ----- Turbulence ----- #
+###########################################################################################################################################################################
+
+diameter = 2.0*opts.radius
+turbLength1 = 0.038*diameter  # Usually in pipeline cases, this is a good estimator. In other case can be used (0.038*hydrDiameter), where hydrDiameter=4*crossSectionArea/wetPerimeter
+turbLength2 = 0.070*diameter  # Based on mixing length formulation. Then different omega equation.
+Ufree = opts.meanVelocity[0] # freestream velocity
+c_mu = opts.c_mu
+useRANS=opts.useRANS
+
+Re=Ufree*diameter/nu_0
+turbI = 0.16*( Re**(-1./8.) )
+kInflow = ((turbI*Ufree)**2) * 3./2. 
+
+if opts.turbLength == 1:
+    dissipationInflow1 = (kInflow**0.5) / (turbLength1) # omega formulation based on turbulent length formulation
+    dissipationInflow = dissipationInflow1
+elif opts.turbLength == 2:
+    dissipationInflow2 = (kInflow**0.5) / ( turbLength2*(c_mu**0.25) ) # omega formulation based on mixing length formulation
+    dissipationInflow = dissipationInflow2
+else:
+    dissipationInflow = (kInflow**0.5) / (opts.scaleLength*opts.water_level)
+
```

Comment:
This version uses a skin friction formula for imposing the turbulence at the pipeline. 
We need to do the same test with the wall function as well!

Reply:
@adimako